zivid_camera/src/settings_generator.cpp

```diff
@@ -80,27 +73,91 @@ void writeToFile(const std::string& file_name, const std::string& text)
   }
 }
 
-std::string convertSettingsPathToZividClassName(const std::string& zivid_settings_class_name, const std::string& path)
+template <typename SettingsRootGroup, typename SettingsNode>
+std::string convertSettingsPathToConfigPath()
 {
-  return "Zivid::" + zivid_settings_class_name + "::" + boost::replace_all_copy<std::string>(path, "/", "::");
-}
+  auto path = std::string(SettingsNode::path);
+  const auto root_path = std::string(SettingsRootGroup::path);
+
+  if (!root_path.empty())
+  {
+    const auto expected_prefix = root_path + "/";
+    if (path.substr(0, expected_prefix.length()) != expected_prefix)
+    {
+      throw std::runtime_error("Expected path '" + path + "' to begin with '" + expected_prefix + "'");
+    }
+    path = path.substr(expected_prefix.length());
+  }
 
-std::string convertSettingsPathToConfigPath(std::string path)
-{
   path = boost::replace_all_copy<std::string>(path, "/", "_");
   const std::regex re("([^_^])([A-Z])");
   path = std::regex_replace(path, re, "$1_$2");  // Convert e.g. ExposureTime to Exposure_Time
   return boost::algorithm::to_lower_copy(path);
 }
 
+template <typename SettingsRootGroup>
+std::string zividSettingsTypeName()
+{
+  if constexpr (std::is_same_v<SettingsRootGroup, Zivid::Settings> ||
+                IsInTuple<SettingsRootGroup, Zivid::Settings::Descendants>::value ||
+                std::is_same_v<SettingsRootGroup, Zivid::Settings::Acquisition> ||
+                IsInTuple<SettingsRootGroup, Zivid::Settings::Acquisition::Descendants>::value)
+  {
+    return Zivid::Settings::name;
+  }
+  else if constexpr (std::is_same_v<SettingsRootGroup, Zivid::Settings2D> ||
+                     IsInTuple<SettingsRootGroup, Zivid::Settings2D::Descendants>::value ||
+                     std::is_same_v<SettingsRootGroup, Zivid::Settings2D::Acquisition> ||
+                     IsInTuple<SettingsRootGroup, Zivid::Settings2D::Acquisition::Descendants>::value)
+  {
+    return Zivid::Settings2D::name;
+  }
+  else
+  {
+    static_assert(DependentFalse<SettingsRootGroup>::value, "Unhandled type.");
+  }
+}
+
+template <typename SettingsNode>
+std::string fullyQualifiedZividTypeName()
+{
+  std::stringstream ss;
+  ss << "Zivid::" + zividSettingsTypeName<SettingsNode>();
+  const auto path = std::string(SettingsNode::path);
+  if (!path.empty())
+  {
+    ss << "::" << boost::replace_all_copy<std::string>(path, "/", "::");
+  }
+  return ss.str();
+}
+
+template <typename SettingsRootGroup>
 class DynamicReconfigureCfgGenerator
 {
 public:
   DynamicReconfigureCfgGenerator(const std::string& class_name) : class_name_(class_name), insert_enabled_(false)
   {
   }
 
-  template <class ValueType>
+  template <typename SettingsNode>
+  auto cfgDefaultValue()
```

Comment:
It's a bit confusing here that the `cfgDefaultValue()` actually gives the min values and false for bools.
  
Perhaps `cfgDummyDefaultValue()` or `cfgTemporaryDefaultValue()` could make it more clear? 
Or a comment in this line:
https://github.com/zivid/zivid-ros/blob/b050280eddb2e44715e5b581c3fddea4d3d233d3/zivid_camera/src/settings_generator.cpp#L235

Reply:
There is already a comment inside of cfgDefaultValue() that explains it, I don't think we should explain it again on the outside. What about renaming method to cfgFileDefaultValue() since this is literally the default value that is set in the .cfg file (which may be changed later by camera-specific default value).

Reply:
Yes, let's go for `cfgFileDefaultValue()`.